### ISSUE TITLE
ST04 fix deletes the space before the outer END when flattening a nested CASE

### ISSUE THREAD

**Issue by devYRPauli** [sqlfluff/sqlfluff#8567]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff fix` with ST04 flattens a nested `CASE` in an `ELSE` clause. When the nested `END` and the outer `END` are on the same line with no comment between them, the fix deletes the space between the nested value and the outer `END`. The fixed SQL does not parse.

The cause is `_nested_end_trailing_comment` in `src/sqlfluff/rules/structure/ST04.py:201-219`. It deletes every whitespace segment on the nested `END` line up to the first comment, then moves that comment to a new line. With no comment on the line, it deletes all the whitespace on the line, including the space before the outer `END`.

The default rule set hides this, because LT01 puts the space back in the same `fix` run. A run with `--rules ST04`, or a config that excludes LT01, keeps the broken output.

### Expected Behaviour

```sql
SELECT CASE WHEN x = 1 THEN a
        WHEN x = 2 THEN b END AS y FROM t
```

### Observed Behaviour

```sql
SELECT CASE WHEN x = 1 THEN a
        WHEN x = 2 THEN bEND AS y FROM t
```

`sqlfluff parse --dialect ansi` on the output reports `Found unparsable section: 'CASE WHEN x = 1 ...'`. When the value is a quoted string, as in `THEN 'b' END END`, the output `THEN 'b'END` still parses, so an identifier or a number before `END` is needed to see the error.

### How to reproduce

```
printf 'SELECT CASE WHEN x = 1 THEN a ELSE CASE WHEN x = 2 THEN b END END AS y FROM t\n' > q.sql
sqlfluff fix --dialect ansi --rules ST04 q.sql
cat q.sql
sqlfluff parse --dialect ansi q.sql
```

### Dialect

ansi. The rule is not dialect-specific.

### Version

sqlfluff 4.3.0 from `main` at `ff76fce3f`, Python 3.12.14.

### Configuration

None. The command line gives the dialect and the rule.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
